### ISSUE TITLE
Make font loading more robust

### ISSUE THREAD

**Issue by jkaessens** [38/plotters#18]
When building the font name database, plotters is extremely sensitive to fonts that it cannot load. Plotters bails with "Error: BackendError(FontError(FontLoadError(UnrecognizedFormat)))" even if the font that I'm actually requesting is already loaded.

In my current case and on a system I don't control, there's an OTF formatted font installed in a system directory that breaks plotters for me. If I run it at my local system and temporarily remove the font in question, plotters works fine.

I was expecting that plotters would just skip loading "unloadable" fonts in the look-up table, instead it panics with the above error. Skipping would at least allow plotters to work with the fonts that are actually loadable.

**Comment by 38** [comment]
Hi, thanks for feedback. That's an good idea and I am happy to add that.

Also could you please provide a stack backtrace to the panic. Since Plotters should returning error code instead of panicking inside the library. So if you can give me the backtrace it will be very helpful.

Thanks again!

**Comment by jkaessens** [comment]
Hi,
now that you put a finger on it, I realize that it's not actually panicking, it's just my lax error handling that made think it was panicking. Sorry for that!

However, the issue is still there. It's `ChartContext::draw()` returning with the error
```rust
Error: BackendError(FontError(FontLoadError(UnrecognizedFormat)))
```
For reference, it's [Nimbus Sans Regular](https://github.com/ArtifexSoftware/urw-base35-fonts/blob/master/fonts/NimbusSans-Regular.otf) that irritates plotters.

**Comment by jkaessens** [comment]
I could narrow it down to the file [src/chart/mesh.rs:157](https://github.com/38/plotters/blob/40e35e1bb0d5ae4e02a4fbb4b8b4211fc90d1e0d/src/chart/mesh.rs#L157) where a default font "Arial" is loaded. If I replace the font face with something that I do have on my system, i.e. "DejaVu Sans", it works as expected. Specifying a different font for the label does not.

**Comment by 38** [comment]
Thanks for digging into it. I guess you are right, and it doesn't support OTF font loading.
The spot you are looking at is for drawing the axis label.

I am going to fix this once I have time. (Currently traveling and no time for coding)
What I am planning to do is allowing customize the default font. (Let me know if this won't help)

Plus I am currently planning to look into the entire font manipulate code since I have realized one of the library I used is leaking memory. I am going to make all those change and cleanup the font manipulating part completely once I have time.

Thanks a lot!
